api/notd/gallery_manager.py

```diff
@@ -176,28 +176,32 @@ async def query_collection_tokens(self, registryAddress: str, limit: int, offset
             .where(orderedListingsQuery.c.tokenListingIndex == 1)
         ).subquery()
         listingsQuery = lowestListingsQuery #if usesListings else LatestTokenListingsTable
+        ownerQuery =(
+            sqlalchemy.select(TokenOwnershipsTable)
+            .where(TokenOwnershipsTable.c.registryAddress == registryAddress)
+        ).subquery()
         if collection.doesSupportErc721:
             query = (
                 sqlalchemy.select(TokenMetadatasTable, TokenCustomizationsTable, listingsQuery, sqlalchemy.literal(1).label('quantity'))
                     .join(TokenCustomizationsTable, sqlalchemy.and_(TokenMetadatasTable.c.registryAddress == TokenCustomizationsTable.c.registryAddress, TokenMetadatasTable.c.tokenId == TokenCustomizationsTable.c.tokenId), isouter=True)
                     .join(TokenOwnershipsTable, sqlalchemy.and_(TokenMetadatasTable.c.registryAddress == TokenOwnershipsTable.c.registryAddress, TokenMetadatasTable.c.tokenId == TokenOwnershipsTable.c.tokenId))
                     .join(listingsQuery, sqlalchemy.and_(TokenMetadatasTable.c.registryAddress == listingsQuery.c.registryAddress, TokenMetadatasTable.c.tokenId == listingsQuery.c.tokenId, listingsQuery.c.offererAddress == TokenOwnershipsTable.c.ownerAddress), isouter=True)
+                    .join(ownerQuery, sqlalchemy.and_(ownerQuery.c.registryAddress == listingsQuery.c.registryAddress, ownerQuery.c.tokenId == listingsQuery.c.tokenId, ownerQuery.c.ownerAddress == listingsQuery.c.offererAddress), isouter=True)
```

Comment:
ownerships table is already used two lines above, why do you add it again? this def looks suspect